napari/layers/base/base.py

```diff
@@ -169,7 +171,9 @@ def __init__(
         self._thumbnail_shape = (32, 32, 4)
         self._thumbnail = np.zeros(self._thumbnail_shape, dtype=np.uint8)
         self._update_properties = True
-        self._name = ''
+        self._name = name
+        self.event_handler = LayerEventHandler(component=self)
+
         self.events = EmitterGroup(
```

Comment:
Is there anyway the actual emitter group can be moved inside the `BaseLayerInterface`? I feel like there is now a 1-1 correspondence between the properties mentioned in the emitter group and the `_on_*_change` methods in the interface, so it could live there. This would then simplify our model files a bit.

Reply:
Yea so I went down this route. The issue becomes the order of initializations. By adding the EmitterGroup instantiation to the Interface, it no longer is "abstract" and needs to be instantiated. It also needs the event_handler callback method on instantiation (or it needs to add it after the fact). But LayerEventHandler get's instantiated with the Layer init method so you can't have access to it until that happens. And since BaseLayerInterface is a superclass of Layer things got weird. Basically I tired adding the event handler info after the init of all the events but then it was confusing how things worked down the line in ImageLayer.  I came to the conclusion that it was better to leave BaseLayerInterface and purely abstract and not have it actually instantiate anything.

Reply:
Ok, that's fine - thanks for looking into it. Maybe at one point it will be possible to say just have a list of properties and then autogenerate a lot of the stuff based on that list :-) but that can come later